test/util/JoinHelpers.h

```diff
@@ -59,31 +59,33 @@ IdTable useJoinFunctionOnIdTables(const IdTableAndJoinColumn& tableA,
  *  `ad_utility::callFixedSize`.
  */
 inline auto makeHashJoinLambda() {
-  return []<int A, int B, int C>(auto&&... args) {
-    return Join::hashJoin(AD_FWD(args)...);
-  };
+  return ad_utility::ApplyAsValueIdentity{
+      [](auto /*valueIdentityA*/, auto /*valueIdentityB*/,
+         auto /*valueIdentityC*/,
+         auto&&... args) { return Join::hashJoin(AD_FWD(args)...); }};
```

Comment:
Interesting, I (@joka921 ) will have a look at why we need those at all (probably for a unified interface).